<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>980</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string notr="true">World Editor</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/Images/WorldBuilder.png</normaloff>:/Images/WorldBuilder.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QQuickWidget" name="quickWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="resizeMode">
       <enum>QQuickWidget::SizeRootObjectToView</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QToolWindowManager" name="toolWidget" native="true">
      <widget class="ContentBrowser" name="contentBrowser" native="true">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>200</width>
         <height>200</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
      </widget>
      <widget class="QWidget" name="propertyWidget" native="true">
       <property name="geometry">
        <rect>
         <x>630</x>
         <y>210</y>
         <width>200</width>
         <height>231</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>2</number>
        </property>
        <property name="leftMargin">
         <number>2</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>2</number>
        </property>
        <property name="bottomMargin">
         <number>2</number>
        </property>
        <item>
         <widget class="QToolButton" name="componentButton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>25</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Adds a new Component to this Actor.</string>
          </property>
          <property name="text">
           <string>Add Component</string>
          </property>
          <property name="popupMode">
           <enum>QToolButton::InstantPopup</enum>
          </property>
          <property name="toolButtonStyle">
           <enum>Qt::ToolButtonTextBesideIcon</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QToolButton" name="commitButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Saves your changes.</string>
            </property>
            <property name="text">
             <string>Commit</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="revertButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Reverts back your unsaved changes.</string>
            </property>
            <property name="text">
             <string>Revert</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="PropertyEditor" name="propertyView"/>
        </item>
       </layout>
      </widget>
      <widget class="SettingsBrowser" name="projectWidget" native="true">
       <property name="geometry">
        <rect>
         <x>-1</x>
         <y>420</y>
         <width>201</width>
         <height>231</height>
        </rect>
       </property>
      </widget>
      <widget class="HierarchyBrowser" name="hierarchy" native="true">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>0</y>
         <width>200</width>
         <height>200</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
      </widget>
      <widget class="ComponentBrowser" name="components" native="true">
       <property name="geometry">
        <rect>
         <x>420</x>
         <y>0</y>
         <width>200</width>
         <height>200</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
      </widget>
      <widget class="Timeline" name="timeline" native="true">
       <property name="geometry">
        <rect>
         <x>420</x>
         <y>210</y>
         <width>201</width>
         <height>201</height>
        </rect>
       </property>
      </widget>
      <widget class="SceneView" name="preview" native="true">
       <property name="geometry">
        <rect>
         <x>420</x>
         <y>420</y>
         <width>201</width>
         <height>231</height>
        </rect>
       </property>
      </widget>
      <widget class="ConsoleManager" name="consoleOutput" native="true">
       <property name="geometry">
        <rect>
         <x>630</x>
         <y>0</y>
         <width>200</width>
         <height>200</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>200</width>
         <height>200</height>
        </size>
       </property>
      </widget>
      <widget class="SettingsBrowser" name="preferencesWidget" native="true">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>420</y>
         <width>201</width>
         <height>231</height>
        </rect>
       </property>
      </widget>
      <widget class="SceneComposer" name="viewportWidget" native="true">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>210</y>
         <width>201</width>
         <height>201</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QTreeView" name="classMapView">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>660</y>
         <width>201</width>
         <height>211</height>
        </rect>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
   <zorder>toolWidget</zorder>
   <zorder>quickWidget</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuBuild_Project">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Build Project</string>
     </property>
     <addaction name="actionBuild_All"/>
     <addaction name="separator"/>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="menuBuild_Project"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionEditor_Mode"/>
    <addaction name="actionGame_Mode"/>
    <addaction name="separator"/>
    <addaction name="actionTake_Screenshot"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Window</string>
    </property>
    <widget class="QMenu" name="menuWorkspace">
     <property name="title">
      <string>Workspace</string>
     </property>
     <addaction name="separator"/>
     <addaction name="actionReset_Workspace"/>
     <addaction name="actionSave_Workspace"/>
    </widget>
    <addaction name="menuWorkspace"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="separator"/>
    <addaction name="actionThunder_Manual"/>
    <addaction name="actionAPI_Reference"/>
    <addaction name="separator"/>
    <addaction name="actionThunder_Answers"/>
    <addaction name="actionReport_Issue"/>
    <addaction name="separator"/>
    <addaction name="actionPlugin_Manager"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuWindow"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+O</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionContent_Browser">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Content Browser</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+B</string>
   </property>
  </action>
  <action name="actionObject_properties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Object properties</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionApplication_Console">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Application Console</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionEditor_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Editor Mode</string>
   </property>
   <property name="shortcut">
    <string notr="true">Esc</string>
   </property>
  </action>
  <action name="actionGame_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Game Mode</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+G</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionScene_objects">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Scene objects</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionComponents">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Components</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionTake_Screenshot">
   <property name="text">
    <string>Take Screenshot</string>
   </property>
   <property name="shortcut">
    <string notr="true">F12</string>
   </property>
  </action>
  <action name="actionPlugin_Manager">
   <property name="text">
    <string>About Plugins...</string>
   </property>
   <property name="toolTip">
    <string>Plugin Manager</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionSave_Workspace">
   <property name="text">
    <string>Save Workspace</string>
   </property>
   <property name="toolTip">
    <string>Save Workspace</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionReset_Workspace">
   <property name="text">
    <string>Reset Workspace</string>
   </property>
   <property name="toolTip">
    <string>Reset Workspace</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>Options...</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionBuild_All">
   <property name="text">
    <string>Build All</string>
   </property>
   <property name="shortcut">
    <string notr="true"/>
   </property>
  </action>
  <action name="actionAPI_Reference">
   <property name="text">
    <string>API Reference</string>
   </property>
  </action>
  <action name="actionReport_Issue">
   <property name="text">
    <string>Report Issue...</string>
   </property>
  </action>
  <action name="actionThunder_Answers">
   <property name="text">
    <string>Thunder Answers</string>
   </property>
  </action>
  <action name="actionThunder_Manual">
   <property name="text">
    <string>Documentation</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string notr="true">Ctrl+N</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QQuickWidget</class>
   <extends>QWidget</extends>
   <header location="global">QtQuickWidgets/QQuickWidget</header>
  </customwidget>
  <customwidget>
   <class>ComponentBrowser</class>
   <extends>QWidget</extends>
   <header>editors/componentbrowser/componentbrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PropertyEditor</class>
   <extends>QTreeView</extends>
   <header>editors/propertyedit/propertyeditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QToolWindowManager</class>
   <extends>QWidget</extends>
   <header>managers/toolwindowmanager/qtoolwindowmanager.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ContentBrowser</class>
   <extends>QWidget</extends>
   <header>editors/contentbrowser/contentbrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>HierarchyBrowser</class>
   <extends>QWidget</extends>
   <header>editors/objecthierarchy/hierarchybrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ConsoleManager</class>
   <extends>QWidget</extends>
   <header>managers/consolemanager/consolemanager.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SceneView</class>
   <extends>QWidget</extends>
   <header>graph/sceneview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Timeline</class>
   <extends>QWidget</extends>
   <header>editors/timeline/timeline.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SettingsBrowser</class>
   <extends>QWidget</extends>
   <header>managers/settingsmanager/settingsbrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SceneComposer</class>
   <extends>QWidget</extends>
   <header>editors/scenecomposer/scenecomposer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
